### PR TITLE
Upgrade to React 18

### LINKED ISSUE #92: Upgrade to React 18
Now the backend has upgraded to React 18, this project should be able too.

### PR DESCRIPTION
**Pull request by sergei-maertens**
Closes #92

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -85,7 +85,7 @@
     "@types/lodash.merge": "^4.6.7",
     "@types/lodash.uniqueid": "^4.0.7",
     "@types/proj4leaflet": "^1.0.10",
-    "@types/react": "^18.0.33",
+    "@types/react": "^18.2.45",
```
Comment (Viicos) [inline package.json:88]: I always find it confusing when types packages' version seems to be aligned with the actual package. Is `@types/react@18.2.x` compatible with `react@18.2.x`? I guess so

File: src/components/builder/prefill/prefill.stories.tsx [hunk 2]
```diff
@@ -60,19 +60,25 @@ export const PluginBlankNoAttributes: Story = {
...
+
+    await waitFor(
+      async () => {
+        await userEvent.keyboard('[Esc]');
+        pluginInput.focus();
+        await userEvent.keyboard('[ArrowDown]');
+        expect(canvas.queryByText('Plugin 1')).toBeInTheDocument();
+        expect(canvas.queryByText('Plugin 2')).toBeInTheDocument();
+      },
+      {timeout: 1500, container: canvasElement}
```
Comment (Viicos) [inline src/components/builder/prefill/prefill.stories.tsx:73]: I hope we will be able to get rid of this arbitrary timeout when moved to jest

File: src/components/builder/registration/registration-attribute.spec.tsx [hunk 3]
```diff
@@ -38,8 +38,10 @@ test('Available registration attributes are retrieved via context', async () =>
 
   // open the dropdown
   const input = await screen.findByLabelText('Registration attribute');
-  input.focus();
-  await user.keyboard('[ArrowDown]');
+  await act(async () => {
+    input.focus();
+    await user.keyboard('[ArrowDown]');
+  });
```
Comment (Viicos) [inline src/components/builder/registration/registration-attribute.spec.tsx:44]: I'm confused: is this `act` call required to suppress the warning? If so, `act` is being called to times, as testing-library apparently handle it already?

**Review by Viicos — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (sergei-maertens) [reply package.json:88]: I think and hope so :grimacing:

File: src/components/builder/prefill/prefill.stories.tsx (hunk 2 above)
Comment (sergei-maertens) [reply src/components/builder/prefill/prefill.stories.tsx:73]: yeah I hope so to, the delays in the getter functions don't seem to reliably work and I have no idea what is or isn't happening. I'm hopeful the jest tests will be better with the mocked timers, which should also result in a faster test suite.

File: src/components/builder/registration/registration-attribute.spec.tsx (hunk 3 above)
Comment (sergei-maertens) [reply src/components/builder/registration/registration-attribute.spec.tsx:44]: I think the `input.focus()` also triggers an internal formik state update, perhaps only that needs to be wrapped in `act` :thinking: 

I'll see if I can move the `user.keyboard` out of the `act` when moving tests to jest.

**Pull request merged**